packages/gamepads_web/lib/gamepad_detector.dart

```diff
@@ -0,0 +1,65 @@
+import 'dart:js_interop';
+
+import 'package:gamepads_platform_interface/api/gamepad_controller.dart';
+import 'package:gamepads_platform_interface/gamepads_platform_interface.dart';
+
+@JS('navigator')
+external Navigator get navigator;
+
+@JS()
+@staticInterop
+class Navigator {}
+
+@JS()
+@staticInterop
+class JSArray {}
+
+@JS()
+@staticInterop
+class Gamepad {}
+
+extension NavigatorGamepads on Navigator {
+  external JSArray getGamepads();
+}
+
+// Extension for JSArray to handle array access
+extension JSArrayInterop on JSArray {
+  external int get length;
+  external JSAny? operator [](int index);
+}
+
+// Extension for Gamepad to access its properties
+extension GamepadInterop on Gamepad {
+  external int get index;
+  external String get id;
+}
+
+List<GamepadController> getGamepads(GamepadsPlatformInterface plugin) {
+  final controllers = <GamepadController>[];
+  final gamepads = navigator.getGamepads();
```

Comment:
Maybe the `getGamepadList` method can be used here instead? And then use a for-in in the loop instead